pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -191,10 +191,12 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	)
 	defer esClient.Close()
 
-	esReachable, err := services.IsServiceReady(d.Client, *externalService)
-	if err != nil {
-		return results.WithError(err)
-	}
+	healthCtx, cancel := context.WithTimeout(context.Background(), esclient.Timeout(d.ES))
+	defer cancel()
+
+	var health esclient.Health
+	health, err = esClient.GetClusterHealth(healthCtx)
+	esReachable := (err == nil && !health.TimedOut)
```

Comment:
This makes sense.  Unfortunately the esClient in `pkg/controller/elasticsearch/client` has no concept of pods, or kubernetes at all, so implementing that logic in that package seems to be out of scope of that particular package.  I could add a method on the default elasticsearch driver in `pkg/controller/es/driver`, which is what I've done, which seems to work well.  Let me know your thoughts on that implementation.

Reply:
Yes, you are right, let's not add anything to `elasticsearch/client`. As I said I think we don't need this additional check in the current form at all.

Reply:
> As I said I think we don't need this additional check in the current form at all.

Right, I removed the referenced form of the check hitting `_cluster/health`, and moved it to checking whether there are any ready pods, which seems to be what you were indicating needs to happen from the below comment.

> The purpose of the esReacheable is just to be able to short-circuit in case there is not even a chance that a Pod will respond. So I think it would be OK to just rely on the readiness property of the Pods here.

Let me know if that is your intent, thanks

Reply:
Ah sorry, I had not seen your latest commit when I wrote that. I will take another look.